src/misc/rwqueue.cpp

```diff
@@ -74,6 +74,53 @@ void RWQueue<T>::Enqueue(T&& item)
 	has_items.notify_one();
 }
 
+template <typename T>
+void RWQueue<T>::BulkEnqueue(std::vector<T>& from_source, const size_t num_requested)
+{
+	// Check that the source can provide the quantity requested
```

Comment:
Maybe assert that `num_requested` is greater than 0?

Reply:
Added; thanks.